synapse/res/providers.json

```diff
@@ -5,13 +5,11 @@
         "endpoints": [
             {
                 "schemes": [
-                    "https://twitter.com/*/status/*",
-                    "https://*.twitter.com/*/status/*",
```

Comment:
Is this removal still necessary? If so, why?

Reply:
It (+ the changes to prefer oEmbed vs. HTML Open Graph) seem to be an unrelated change to what is proposed in the description.

Reply:
Sites listed in providers.json will use the specified oEmbed url and completely bypass opengraph parsing of the page being previewed.

Since this change is to fix previously broken opengraph previews, we should likewise remove these oEmbed overrides for the pages that we fixed.

Reply:
@clokep  removed the prefer change from this PR since that is not related.

Reply:
> @clokep removed the prefer change from this PR since that is not related.

Sorry, what did you remove? I can still see this change in the PR's diff.

Reply:
Removed the change on line 330 https://github.com/matrix-org/synapse/pull/11985/files#diff-bab0f9e113994dc8e6d6dedc4e2d3ce8b5ff6e059dc181ccf4db4c3358d60b29R330

I believe the providers change should be part of this PR since the purpose is to fix the opengraph fetching.  This providers list will bypass the change completely.

Reply:
I removed the change on line 330 from this PR.  This PR now changes only the user agent and providers.json, and is unrelated to #11669 now.

The change to providers.json is necessary specifically so twitter urls will use opengraph instead of bypassing the logic.